File: app/src/main/java/com/weathercock/cleanarchitecture/presentation/presenter/activity/AboutActivityViewModel.java
package com.weathercock.cleanarchitecture.presentation.presenter.activity;

import android.content.Context;

import com.weathercock.cleanarchitecture.presentation.listener.presenter.activity.AboutActivityViewModelListener;

/**
 * About Activity ViewModel
 *
 * @author Kensuke
 */
public class AboutActivityViewModel {

    private Context context;
    private AboutActivityViewModelListener aboutActivityViewModelListener;

    public AboutActivityViewModel(Context context, AboutActivityViewModelListener aboutActivityViewModelListener) {
        this.context = context;
        this.aboutActivityViewModelListener = aboutActivityViewModelListener;
    }

    public void onClickBackIcon() {
        aboutActivityViewModelListener.onPerformFinish();
    }

}

File: presentation/src/main/java/com/kazakago/cleanarchitecture/presentation/viewmodel/activity/AboutActivityViewModel.kt
package com.kazakago.cleanarchitecture.presentation.viewmodel.activity

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.kazakago.cleanarchitecture.presentation.listener.activity.AboutActivityViewModelListener

class AboutActivityViewModel(application: Application) : AndroidViewModel(application) {

    class Factory(private val application: Application) : ViewModelProvider.Factory {
        @Suppress("UNCHECKED_CAST")
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return AboutActivityViewModel(application) as T
        }
    }

    var listener: AboutActivityViewModelListener? = null

    fun onClickBackIcon() {
        listener?.finish()
    }

}
